tests/get_or_create/tests.py

```diff
@@ -503,6 +505,45 @@ def raise_exception():
         )
         self.assertFalse(created)
 
+    def test_update_only_defaults_and_auto_now_fields_when_local_fields(self):
+        publisher = Publisher.objects.create(name="Acme Publishing")
+        book = Book.objects.create(publisher=publisher, name="The Book of Ed & Fred")
+
+        with CaptureQueriesContext(connection) as captured_queries:
+            _, created = Book.objects.update_or_create(
+                pk=book.pk,
+                defaults={"publisher": publisher},
+            )
+
+            # Retry with _id field
+            _, created_with_id_field = Book.objects.update_or_create(
+                pk=book.pk,
+                defaults={"publisher_id": publisher.id},
+            )
+
+        self.assertFalse(created)
+        self.assertFalse(created_with_id_field)
+        update_queries = [
+            q["sql"] for q in captured_queries if q["sql"].startswith("UPDATE")
+        ]
+        # Ensure that we execute UPDATE twice
+        self.assertEqual(len(update_queries), 2)
+        for query in update_queries:
+            self.assertIsNotNone(query)
+            self.assertIn("publisher_id_column", query)
+            self.assertIn("updated", query)
+            # Name should not get updated
+            self.assertNotIn("name", query)
+
+    def test_mti_update_non_local_concrete_fields(self):
+        journalist = Journalist.objects.create(name="Jane", specialty="Politics")
+        journalist, created = Journalist.objects.update_or_create(
+            pk=journalist.pk,
+            defaults={"name": "John"},
+        )
+        self.assertFalse(created)
+        self.assertEqual(journalist.name, "John")
```

Comment:
`test_mti_update_non_local_concrete_fields` works without the patch, so we should move it to a separate commit.

Reply:
[8a8ed41](https://github.com/django/django/pull/16072/commits/8a8ed41b87e6031d9a5d28ea82156753a2defd13)